src/arithmetic/algebraic_expression.h

```diff
@@ -35,10 +41,12 @@ struct AlgebraicExpression {
         struct {
             bool diagonal;          // Diagonal matrix.
 		    GrB_Matrix matrix;      // Matrix operand.
+		    GrB_Vector vector;      // Vector operand.
```

Comment:
While these are both only 8-byte pointers, they should probably be joined in a union accessed based on `type`.